### ISSUE TITLE
False positive on the `Lint/ReturnInVoidContext` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12386]
## Actual behavior

I have the following code:
```ruby
class Notifications::CreateInteractor < ApplicationInteractor
  def data=(value)
    return @data = nil if value.blank?

    @data = value
  end

  # ...
end
```

I run rubocop and get offences:
```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop app/interactors/notifications/create_interactor.rb 
Inspecting 1 file
W

Offenses:

app/interactors/notifications/create_interactor.rb:5:5: W: Lint/ReturnInVoidContext: Do not return a value in data=.
    return @data = '' if value.nil?
    ^^^^^^

1 file inspected, 1 offense detected
```

## Rubocop

```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.57.2 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.19.0
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.22.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.25.0
  - rubocop-thread_safety 0.5.1
```

**Comment by koic** [comment]
It appears that using `return` is not expected in terms of code semantics. It should be updated to a style that does not use `return`.

**Comment by ydakuka** [comment]
Hm, ok, but the following code looks pretty weird:

```ruby
class Notifications::CreateInteractor < ApplicationInteractor
  def data=(value)
    if value.blank?
      @data = nil

      return
    end

    @data = value
  end

  # ...
end
```
